### PR TITLE
Add several supporting lua functions.

### PR DESCRIPTION
**Pull request by postwait**
* mtev.get_ipv4 (get local IP address based on target route)
 * mtev.uname (return uname(2) details)
 * mtev.reverse_details (show details of a reverse connection)

### CODE REVIEW

File: src/modules/lua.c [hunk 1]
```diff
@@ -1316,6 +1316,32 @@ mtev_lua_open(const char *module_name, void *lmc,
   return L;
 }
 
+int
+mtev_lua_push_inet_ntop(lua_State *L, struct sockaddr *r) {
+  char remote_str[128];
```
Comment (cody271) [inline src/modules/lua.c]: Does `inet_ntop()` NULL terminate safely? man page isn't clear..

File: src/modules/lua.c [hunk 2]
```diff
@@ -1316,6 +1316,32 @@ mtev_lua_open(const char *module_name, void *lmc,
...
+  int len;
+  switch(r->sa_family) {
+    case AF_INET:
+      len = sizeof(struct sockaddr_in);
+      inet_ntop(AF_INET, &((struct sockaddr_in *)r)->sin_addr,
```
Comment (cody271) [inline src/modules/lua.c]: Needs `errno` check.

File: src/modules/lua.c [hunk 3]
```diff
@@ -1316,6 +1316,32 @@ mtev_lua_open(const char *module_name, void *lmc,
...
+                remote_str, len);
+      lua_pushstring(L, remote_str);
+      lua_pushinteger(L, ntohs(((struct sockaddr_in *)r)->sin_port));
+      break;
+    case AF_INET6:
+      len = sizeof(struct sockaddr_in6);
+      inet_ntop(AF_INET6, &((struct sockaddr_in6 *)r)->sin6_addr,
```
Comment (cody271) [inline src/modules/lua.c]: Needs `errno` check.

File: src/mtev_reverse_socket.c [hunk 4]
```diff
@@ -2237,6 +2256,29 @@ mtev_reverse_socket_connection_shutdown(const char *address, int port) {
   pthread_mutex_unlock(&reverses_lock);
   return success;
 }
+mtev_boolean
+mtev_connection_do(const char *address, int port, void (*cb)(mtev_connection_ctx_t *, reverse_socket_t *, void *), void *closure) {
+  mtev_hash_iter iter = MTEV_HASH_ITER_ZERO;
+  mtev_boolean invoked = mtev_false;
+  char remote_str[INET6_ADDRSTRLEN + 1 + 5 + 1];
```
Comment (cody271) [inline src/mtev_reverse_socket.c]: What is `INET6_ADDRSTRLEN + 1 + 5 + 1` from ?

File: src/mtev_reverse_socket.c [hunk 5]
```diff
@@ -179,15 +179,34 @@ typedef struct {
...
-typedef struct {
+struct reverse_socket {
   reverse_socket_data_t data;
   char *id;
   pthread_mutex_t lock;
   uint32_t refcnt;
-} reverse_socket_t;
+};
+
+#undef RSACCESS
```
Comment (cody271) [inline src/mtev_reverse_socket.c]: These macros are all already in the header file ?

File: src/modules/lua.c (hunk 1 above)
Comment (postwait) [reply src/modules/lua.c]: yes

File: src/modules/lua.c (hunk 2 above)
Comment (postwait) [reply src/modules/lua.c]: this is a pre-existing function... and we can't get an error under this usage condition.,

File: src/modules/lua.c (hunk 3 above)
Comment (postwait) [reply src/modules/lua.c]: same

File: src/mtev_reverse_socket.c (hunk 4 above)
Comment (postwait) [reply src/mtev_reverse_socket.c]: longest name + : + longest port + \0

**Pull request merged**